src/lib/Server/Output/ValueObjectVisitor/Location.php

```diff
@@ -147,7 +156,27 @@ protected function visitLocationAttributes(Visitor $visitor, Generator $generato
             )
         );
         $generator->endAttribute('href');
-        $visitor->visitValueObject(new RestContentValue($location->contentInfo));
+
+        $content = $location->getContent();
+        $contentInfo = $location->contentInfo;
+
+        try {
+            $mainLocation = $contentInfo->mainLocationId === $location->id
+                ? $location
+                : $this->locationService->loadLocation($contentInfo->mainLocationId);
+        } catch (NotFoundException | UnauthorizedException $e) {
```

Comment:
Does `NotFoundException` make sense here? I get it why we're preventing the `UnauthorizedException` as it would be "normal flow" of operations, but if we cannot load Location for Content it seems we would cover up an issue with data.